_posts/2017/2017-02-20-continuous-documentation-with-ansible.md

```diff
@@ -0,0 +1,207 @@
+
+This post will outline how [Ansible](https://www.ansible.com/) can help in approaching these challenges.
+It also includes a short discussion on the relation of Ansible to other tools supporting continuous delivery and
+scalability of the development process, [Docker](https://www.docker.com/) and [Jenkins](https://jenkins.io/).
+
+# Documentation vs. Automation
+The way we think about documentation [is changing](https://devops.com/documenting-devops-agile-automation-and-continuous-documentation/).  
+For the documentation of application software it is generally accepted that it should be generated from code, to leave as little possibility for deviation as possible.  
+In the case of "management code" this is more difficult, because the things being modelled are often heterogeneous and not as focussed in terms of their implementation as for example a REST-API or a library / module.  
+The concept of [using declarative formats for automation](https://12factor.net/#introduction) can fill this gap.
+
+# Structure of Ansible projects
+As there are many sources of Ansible best practices to be found on the web, instead of writing another one, I would like to assume the viewpoint of someone reading a Ansible code, rather than writing it.
+Note that I am using the term "Ansible project" without implying a certain structure in terms of source code repositories.  
+
+## Playbooks
+[Playbooks](http://docs.ansible.com/ansible/playbooks.html) are the entry point when looking at an Ansible project.  
+A common convention is two have two kinds of Playbooks.
+
+### Setup Playbooks
+One type of playbook would initially provision / prepare an environment to deploy an application into it.  
+These playbooks would typically be named `setup-<project>.yml`.
+While this type could traditionally be seen more on the IT department's side of an organization, this type of playbook would also be used [to set up a development environment](https://github.com/ePages-de/mac-dev-setup) (i.e. developer laptop) for that application.  
+The point of using the same technique for both of these things is that it becomes easier to manage and document differences between them in a joint effort. This continually provides value for the company, because setup times for developers are reduced.  
+Using playbooks to install a development environment is a good way of incorporating new developers. Invariably occurring problems are an opportunity to familiarize with the environment and getting used to the workflow by documenting their solution.
+
+### Deployment Playbooks
+The other kind of playbooks would be named `deploy-<project>.yml`.  
+These are meant to execute continuously, deploying changing versions of an application.  
+This type of playbook would typically be executed during a CI/CD run, and a failure during that phase could be the cause for investigating the playbook in the first place.
+
+Let's look at an example playbook (`deploy-unity.yml`):
```

Comment:
Is this is "real" example from our environment? I don't know many insights we give an external reader here into our environment. But I would prefer a "fake" example.